From 9c56e3baa9d87928ed983ca6dfcf6c982c5e2111 Mon Sep 17 00:00:00 2001
From: Richard Hu <richard.hu@technexion.com>
Date: Thu, 5 Jul 2018 11:18:24 +0800
Subject: [PATCH] optee-os-imx: fix build failure when machine isn't imx8mqevk

Modify the condition to check SOC type instead of board type.
If the SOC is i.mx8m, then apply 'mx8mqevk' as PLATFORM_FLAVOR to build optee-os.
---
 imx/meta-bsp/recipes-security/optee-imx/optee-os-imx_git.bb | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/imx/meta-bsp/recipes-security/optee-imx/optee-os-imx_git.bb b/imx/meta-bsp/recipes-security/optee-imx/optee-os-imx_git.bb
index 9455eac..3e4fb62 100644
--- a/imx/meta-bsp/recipes-security/optee-imx/optee-os-imx_git.bb
+++ b/imx/meta-bsp/recipes-security/optee-imx/optee-os-imx_git.bb
@@ -19,6 +19,7 @@ B = "${WORKDIR}/build.${OPTEE_PLATFORM}"
 
 python () {
 	machine = d.getVar("MACHINE", True)
+	machineoverrides=(d.getVar('MACHINEOVERRIDES', True) or '').split(':')
 
 	import re
 	if re.match('imx6qpdlsolox',machine):
@@ -27,8 +28,8 @@ python () {
 		subplatform = 'mx6ulevk'
 	elif re.match('imx6ull',machine):
 		subplatform = 'mx6ullevk'
-	elif re.match('imx',machine):
-		subplatform = machine[1:]
+	elif 'mx8mq' in machineoverrides:
+		subplatform = 'mx8mqevk'
 	else:
         	bb.fatal("optee-os-imx doesn't recognize this MACHINE")
 	d.setVar("OPTEE_PLATFORM", subplatform)
-- 
1.9.1

